### PR TITLE
v7

### PR DESCRIPTION
**Pull request by wa-rren-dev**
## Description

This is a fast-moving and relatively cavalier proposal branch for version 7 of the `nhsuk-frontend` package. Use at your own risk!

See the branch's [changelog](https://github.com/nhsuk/nhsuk-frontend/blob/alpha-v7/CHANGELOG.md) for info on what's changing.

## Before release

- [x] Write tests for Tabs
- [x] Write tests for Character Count
- [x] Experimental label on the frontend demo

### CODE REVIEW

**Comment by kevinkuszyk** [comment]
@wa-rren-dev do you have an ETA for shipping the final v7, and any idea of milestones along the way?

I'm looking forward to no longer seeing all the SASS errors when we build our app, and I'll also try and keep [NHSDigital/nhsuk-react-components](https://github.com/NHSDigital/nhsuk-react-components) in sync if you have a roadmap you can publish.

Thanks, and as always, keep up the good work!

**Comment by wa-rren-dev** [comment]
Cheers @kevinkuszyk, we don't have a release plan just yet as we're running with a skeleton crew at the moment, but the alpha is an attempt to get some momentum going. The main barrier to getting it released will be the production of the guidelines to go in the service manual for the new components, but if it comes to it we could possibly release a more reliable beta version of v7 with assurances that there won't be any sweeping changes.

**Comment by andymantell** [comment]
Hi Adam,

1. Good question. It was grey originally and I switched it because it "looked funny" in the demos. But that's not really a great reason, and I do note the govuk one is grey. See before and after below. What do you think, should we put it back to grey? Or maybe make grey the default but have a modifier which gives you white tabs, in case that makes more sense in whatever context people put it in?

Now that I've written all that, I'm leaning towards just making it grey - to match the govuk one as you say. Can I get another opinion @wa-rren-dev ?

image

2. It should be left aligned - that's a bug in the version you've got there. Have you updated recently? This should have been fixed by 336c98ae3869abc94ffde1222a474f812bfc3f95

**Comment by andymantell** [comment]
Sorry, the actual commit that fixed the tab alignment was this:

https://github.com/nhsuk/nhsuk-frontend/pull/831/commits/43f4fa5f901568c50619e4d26be48ec6a1158c62#diff-92bf1bf018dd660a48623539ce33dd98f218877144e4f6beb606693c46e8b4fa

**Comment by chrimesdev** [comment]
Thanks @andymantell I can see what you mean. I think white Cards (and other colored components) can often be overused on NHS pages, as they stand out more, so it might convey more importance than the content on the rest of the page as I think you're more drawn to it. I'll speak to our designer tomorrow and see what he thinks. 

If it helps here's a screenshot of a Figma design with the Tabs with a grey background in a real scenario:

Screenshot 2022-08-11 at 20 42 42

**Comment by wa-rren-dev** [comment]
Regarding the tab backgrounds - I wouldn't like to make any particular suggestion! It really needs design and UR around it. 

That said, my personal preference would be to keep the tabs white. Without the background it can be hard when scanning to see what is part of a tab's content and what is proceeding page content.

Hopefully this sort of thing will be picked up in the forthcoming tabs work that mission team 1 are planning on doing. Have referenced this thread in the [assosciated epic](https://github.com/nhsuk/nhsuk-service-manual/issues/1790).

**Comment by chrimesdev** [comment]
Thanks @andymantell and @wa-rren-dev! We've just gone live with the Tabs component (https://www.demo.livi.thirdparty.nhs.uk/services/order-a-repeat-prescription/) will let you know how it goes!

**Comment by wa-rren-dev** [comment]
> Thanks @andymantell and @wa-rren-dev! We've just gone live with the Tabs component (https://www.demo.livi.thirdparty.nhs.uk/services/order-a-repeat-prescription/) will let you know how it goes!

Great, any feedback you get do send it back round if you can!

**Comment by chrimesdev** [comment]
Is the `[7.0.0-alpha.2](https://www.npmjs.com/package/nhsuk-frontend/v/7.0.0-alpha.2)` package ready/safe to be used in production services? Is it being used on any other live services currently?

**Comment by wa-rren-dev** [comment]
Hi @chrimesdev, not being used on any live services that I'm aware of. Based on the pace of change so far I wouldn't recommend using it in production unless you're happy to pin the alpha version you're using in your dependencies. I really can't imagine anything else breaking at this point but based on the possibility that this alpha might outlive all of us, I wouldn't let it auto install the latest version. We're trying to retain the ability to still make sweeping changes at this point.

**Review by mcheung-nhs — CHANGES_REQUESTED** [review]

File: packages/components/warning-callout/README.md [hunk 1]
```diff
@@ -44,11 +49,14 @@ If you’re using Nunjucks macros in production be aware that using `html` argum
 <div class="nhsuk-warning-callout">
   <h3 class="nhsuk-warning-callout__label">
     <span role="text">
-      <span class="nhsuk-u-visually-hidden">Important: </span>
+      <span class="nhsuk-u-visually-hidden">Important:</span>
```
[the remark this answers is not in the document]
Comment (andymantell) [reply packages/components/warning-callout/README.md:52]: nbsp maybe?

**Review by andymantell — CHANGES_REQUESTED** [review]
Steal prettier config from here: https://github.com/alphagov/govuk-frontend/blob/main/.prettierrc.js

File: packages/components/tables/README.md [hunk 2]
```diff
@@ -43,47 +43,10 @@ Find out more about the table component and when to use it in the [NHS digital s
...
-      {
-        text: "Itchy blisters"
-      },
-      {
-        text: "shingles, chickenpox"
-      }
-    ]
-  ]
-}) }}
+{% from 'components/tables/macro.njk' import table %} {{ table({ panel: false, caption: "Skin symptoms and possible
```
[the remark this answers is not in the document]
Comment (andymantell) [reply packages/components/tables/README.md:46]: Agreed - I am going to revert markdown files to their master state, prevent prettier from formatting embedded code examples and then reformat them. This one is a bit nasty - it's trying to format it as html which it isn't

File: docs/installation/installing-compiled.md [hunk 3]
```diff
@@ -14,44 +14,44 @@ If you require any of this functionality, you should [install using npm](/docs/i
...
-    in the example below, to the version you are using.
+   Make sure you change the version numbers of the `[latest version].min.css` and `[latest version].min.js` files,
+   in the example below, to the version you are using.
 
-    ```html
-    <!-- Styles -->
-    <link rel="stylesheet" href="css/nhsuk-[latest version].min.css">
+   ```html
+   <!-- Styles -->
+   <link rel="stylesheet" href="css/nhsuk-[latest version].min.css" />
```
[the remark this answers is not in the document]
Comment (andymantell) [reply docs/installation/installing-compiled.md]: have reverted this in the markdown files

File: packages/components/tabs/template.njk [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{#- If an id 'prefix' is not passed, fall back to using the name attribute
+   instead. We need this for error messages and hints as well -#}
+{% set idPrefix = params.idPrefix if params.idPrefix -%}
+
+<div {%- if params.id %} id="{{params.id}}"{% endif %} class="nhsuk-tabs {%- if params.classes %} {{ params.classes }}{% endif %}" {%- for attribute, value in params.attributes %} {{attribute}}="{{value}}"{% endfor %} data-module="nhsuk-tabs">
+  <h2 class="nhsuk-tabs__title">
```
[the remark this answers is not in the document]
Comment (andymantell) [reply packages/components/tabs/template.njk:6]: Agreed, would be a good enhancement

File: packages/components/warning-callout/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (andymantell) [reply packages/components/warning-callout/README.md]: Space has been reinstated

File: packages/components/character-count/README.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+| **errorMessage**         | object | No       | Arguments for the error message component (e.g. text). See [error message](https://github.com/nhsuk/nhsuk-frontend/tree/master/packages/components/error-message) component.                                                                                                                                                                              |
+| **classes**              | string | No       | Optional additional classes to add to the textarea tag. Separate each class with a space.                                                                                                                                                                                                                                                                 |
+| **autocomplete**         | string | No       | Attribute to [identify input purpose](https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose.html), for instance "postal-code" or "username". See [Autofilling form controls: the autocomplete attribute](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill) for the full list of attributes that can be used. |
```
[the remark this answers is not in the document]
Comment (andymantell) [reply packages/components/character-count/README.md]: No that's slipped in by mistake from the textarea docs, I'll remove it

File: app/components/all.njk [hunk 6]
```diff
@@ -3,6 +3,7 @@
 {% from 'components/action-link/macro.njk' import actionLink %}
 {% from 'components/breadcrumb/macro.njk' import breadcrumb %}
 {% from 'components/card/macro.njk' import card %}
+{% from 'components/character-count/macro.njk' import characterCount %}
 {% from 'components/contents-list/macro.njk' import contentsList %}
```
[the remark this answers is not in the document]
Comment (andymantell) [reply app/components/all.njk:7]: Yeah good shout - done.

**Review by andymantell — DISMISSED** [review]

**Review by mcheung-nhs — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,23 +1,70 @@
...
+
+- Rework clickable cards using CSS invisible overlay rather than JS event handler to avoid problems with using Ctrl-click, middle click, right click to open new tabs ([PR 762](https://github.com/nhsuk/nhsuk-frontend/pull/762)).  
+  This change removes the `Card` component's associated JavaScript. If you are bundling this yourself in your pipeline you will need to edit your JS entrypoint to remove the lines which read: `import Card from '[wherever]';` and `Card();`
+  You do not need to update your html as part of this change - this remains the same.
+
+:wrench: **Maintenance**
+
+- Removes support for LibSass and Ruby Sass in favour of [Dart Sass](https://www.npmjs.com/package/sass)
+- Upgrade all dependencies to their latest versions where possible
+- Pinned the Node version to the latest LTS version at the time of publishing (16.16.0). In environments where Volta is installed the appropriate version will be used
```
[the remark this answers is not in the document]
Comment (andymantell) [reply CHANGELOG.md]: Resolved - I linked to Prettier but removed Volta as discussed.

**Review by mcheung-nhs — APPROVED** [review]

**Pull request merged**